src/mbgl/util/offscreen_texture.cpp

```diff
@@ -36,15 +39,31 @@ class OffscreenTexture::Impl {
         return size;
     }
 
+    void bindRenderbuffers() {
+        if (!framebuffer) {
+            texture = context.createTexture(size, gl::TextureFormat::RGBA, unit);
+            colorTarget = context.createRenderbuffer<gl::RenderbufferType::RGBA4>(size);
+            depthTarget = context.createRenderbuffer<gl::RenderbufferType::DepthComponent>(size);
+            framebuffer = context.createFramebuffer(*colorTarget, *depthTarget, *texture);
+        } else {
+            context.bindFramebuffer = framebuffer->framebuffer;
+        }
+
+        context.viewport = { 0, 0, size };
+    }
+
 private:
     gl::Context& context;
     const Size size;
+    gl::TextureUnit unit;
     optional<gl::Framebuffer> framebuffer;
     optional<gl::Texture> texture;
+    optional<gl::Renderbuffer<gl::RenderbufferType::RGBA4>> colorTarget;
+    optional<gl::Renderbuffer<gl::RenderbufferType::DepthComponent>> depthTarget;
 };
 
-OffscreenTexture::OffscreenTexture(gl::Context& context, const Size size)
-    : impl(std::make_unique<Impl>(context, std::move(size))) {
+OffscreenTexture::OffscreenTexture(gl::Context& context, const Size size, gl::TextureUnit unit)
+    : impl(std::make_unique<Impl>(context, std::move(size), unit)) {
```

Comment:
Instead of adding the `unit` parameter to the constructor (where it isn't used), we should add it to the `bind` call to be in line with other uses of this parameter. The texture unit is a render-time property, and not a property of the texture object.